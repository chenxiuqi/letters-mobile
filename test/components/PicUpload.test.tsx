import React from "react";
import { PicUpload } from "@components";
import renderer from "react-test-renderer";
import { Image, TouchableOpacity } from "react-native";

const setup = () => {
  const element = renderer.create(<PicUpload />);
  return {
    element,
  };
};

describe("PicUpload component", () => {
  it("should render", () => {
    const { element } = setup();
    const tree = element.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should display blank with no image", () => {
    const { element } = setup();
    const picUpload = element.root;
    const touchable = picUpload.findByType(TouchableOpacity);
    expect(touchable).toBeDefined();
    const image = picUpload.findAllByType(Image);
    expect(image.length).toBe(0);
  });
});
