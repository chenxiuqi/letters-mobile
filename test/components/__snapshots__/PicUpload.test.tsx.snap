// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PicUpload component should render 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#3490dc",
      "borderRadius": 50,
      "height": 100,
      "opacity": 1,
      "overflow": "hidden",
      "width": 100,
    }
  }
/>
`;
